src/main/java/CarPosition.java

```diff
@@ -0,0 +1,19 @@
+public class CarPosition {
+
+    private int position;
+
+    public CarPosition(int position){
+        if(position < 0){
+            throw new IllegalArgumentException("차 움직임은 음수일 수 없습니다.");
+        }
+        this.position = position;
+    }
+
+    public CarPosition move(){
+        return new CarPosition(this.position+1);
```

Comment:
```suggestion
        return new CarPosition(this.position + 1);
```

불변을 지키려고 하는것은 좋지만, 계속 인스턴스를 생성한다면 메모리 성능적으로 계속 추가될거 같아요.
이게 불변과 성능간의 트레이드 오프가 되는데요.한번 고민해보시면 좋을거 같아요.